app/controllers/posts_controller.rb

```diff
@@ -1,8 +1,8 @@
 class PostsController < ApplicationController
   skip_before_action :verify_authenticity_token
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = current_user
+    @posts = Post.order(created_at: :desc).where(author_id: current_user.id)
   end
```

Comment:
- [ ] solves the N+1 problem when fetching all posts and their comments, you are missing include keyword to solve the n+1 issue. use this[ link](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh) for more explanation.

Reply:
Thank you for letting me know. I fixed it. 👍